src/pharmpy/basic/matrix.py

```diff
@@ -144,6 +144,23 @@ def evalf(self, d: Mapping) -> np.ndarray:
     def deserialize(cls, s) -> Matrix:
         return cls(sympy.parse_expr(s))
 
+    def is_square(self) -> bool:
+        return self.rows == self.cols
+
+    def is_symmetric(self) -> bool:
+        return self.is_square() and all(
+            self[i, j] == self[j, i] for i in range(self.rows) for j in range(i)
+        )
```

Comment:
Symengine has an 'is_symmetric` method that could be used instead. This will also check if the matrix is square (necessary to be symmetric).